src/cache_invalidate.c

```diff
@@ -77,16 +78,27 @@ cache_invalidate_callback(Datum arg, Oid relid)
 	if (!ts_extension_is_loaded())
 		return;
 
-	catalog = ts_catalog_get();
+	/* The cache invalidation can be called indirectly further down in the
+	 * call chain by calling `get_namespace_oid`, which can trigger a
+	 * recursive cache invalidation callback. To prevent infinite recursion,
+	 * we stop it here and instead consider the cache as invalidated, which
+	 * will allow the `get_namespace_oid` to retrieve the OID from the heap
+	 * table and return it properly. */
+	if (!in_recursion)
+	{
+		in_recursion = true;
+		catalog = ts_catalog_get();
```

Comment:
Nit: you can now move the `Catalog` declaration inside the local context here.

Reply:
Moved.